### PR TITLE
Fix tab expansion for `Get-Process` on macOS

### LINKED ISSUE #7174: No-argument tab-completion fails with Get-Process on macOS, possibly due to process objects lacking a ProcessName property
Steps to reproduce
------------------

On macOS, try the following:

```powershell
Get-Process  # < -- press <tab> here (whether once or multiple times makes no difference)
```

On Windows and Linux you get the names of all running processes, but on macOS nothing happens.

Expected behavior
-----------------

A  list (menu) of the names of all running processes.

Actual behavior
---------------

Nothing happens at all.

The problem goes away once you've typed at least prefix char. of the process name to match.

_Presumed_ cause:

On macOS, many processes output by `Get-Process` lack a `.Name` aka `.ProcessName` value.

As an aside: Is that expected or a bug in itself?

It seems that these entries without a `.Name` value cause the problem when the list of completions is constructed:

@lzybkr provided the following code to diagnose the problem, which simulates what happens when you tab-complete, while also surfacing exceptions:

```powershell
$line = "Get-Process "  # Simulate `Get-Process <tab>`
TabExpansion2 $line $line.Length
```

The above, rather than outputting a `[System.Management.Automation.CommandCompletion]` object with all  process names, reports the following error:

```
Exception calling "CompleteInput" with "3" argument(s): "Cannot process argument because the value of argument "listItemText" is null. Change the value of argument "listItemText"
...
```

Environment data
----------------

```powershell
PowerShell Core v6.1.0-preview.3 on macOS 10.13.5
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

macOS has a number of nameless processes.  When tab expansion gets called, the empty process names results in an exception which fails the entire completer.  Fix is to skip processes with empty name so that the named processes still show up in the result list.

Fix https://github.com/PowerShell/PowerShell/issues/7174

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -3172,6 +3172,11 @@ private static void NativeCompletionProcessCommands(CompletionContext context, s
                         completionText = quote + completionText + quote;
                     }
 
+                    if (string.IsNullOrEmpty(listItemText))
+                    {
+                        continue;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: If we now silently skip empty strings should we document this?

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs (hunk 1 above)
Comment (lzybkr) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: I'm surprised a process might not have a name - is this patching over a problem elsewhere, like in .Net?

At any rate, a comment on why this is expected would be useful.

Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: @lzybkr if I run PowerShell with `sudo`, the process names are there.  So presumably these are system processes where some properties are protected.  I'll add a comment.

**Review by BrucePay — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**